### DISCUSSION TITLE
Install templates globally for user, not locally for every project

### DISCUSSION THREAD

**Discussion by venpopov** [quarto-dev/quarto-cli#8469] [Feature Requests]
### Description

Currently quarto templates/extensions need to be installed anew in every project folder that uses them. It would be much more convenient if we can install them in a central folder and be able to access them regardless of where the notebook resides. Just like R packages are centrally stored for the user and we don't have to install them anew for every RProject and keep a copy in the working directory.

For example, quarto templates can be install under the user home folder ~quarto/templates, and then when rendering them quarto looks for matching templates there.

**Comment by mcanouil** [comment]
That's by design and for reproducibility.
If extensions are global, projects would no longer be portable/shareable.
You may have notice that there is no extension repository or manager which would require man power to maintain it.

**Reply by venpopov** [reply]
I understand the reasoning, but there should be easy ways to deal with that. For example, we could specify the extension version tagged on github, and when the file is rendered on a different machine, the extension is installed and loaded - or add the extension as a dependency. My point is not to make it less reproducible, but rather to avoid manual duplication of effort - the whole point of all these cools is to make life easier and code reproducible. 

How about this option - there is a central user library for extensions (I mean on a user's machine, with their installed extensions, not an extension repository), and whenever an extension is used in a quarto file, during the rendering instead of throwing an error, there is prompt to install it - just like you have to do now manually.

**Reply by mcanouil** [reply]
You can use hard links if you want.

> For example, we could specify the extension version tagged on github, and when the file is rendered on a different machine, the extension is installed and loaded - or add the extension as a dependency.

This is not as easy as you think.
And this is not as easy for users as you think.
They would need to take care of version while current extension mechanism install from main branch not from tags.
So, it would means a lot of work from user.
Yes, duplication is not great but here it means one can clone a repository, install Quarto and nothing more is needed to render the project.


Note that this does not mean, it won't happen, simply not in a foreseeable future as there are priorities on other and a lot of things.

**Reply by venpopov** [reply]
Thank you for the explanation.
